nilearn/regions/tests/test_signal_extraction.py

```diff
@@ -66,6 +70,86 @@ def test_generate_labeled_regions():
     assert (len(np.unique(get_data(regions))) == n_regions + 1)
 
 
+def test_shape_affine_check_with_labels():
+    """Test to ensure correct check for valid shapes & affines of labels."""
+
+    shape = (8, 9, 10)
+    test_shape = (8, 9, 5)
+
+    # data
+    affine = np.eye(4)
+    test_affine = np.eye(3)
+
+    # labels
+    labels_data = np.zeros(shape, dtype=int)
+    h0 = shape[0] // 2
+    h1 = shape[1] // 2
+    h2 = shape[2] // 2
+    labels_data[:h0, :h1, :h2] = 1
+    labels_data[:h0, :h1, h2:] = 2
+    labels_data[:h0, h1:, :h2] = 3
+    labels_data[:h0, h1:, h2:] = 4
+    labels_data[h0:, :h1, :h2] = 5
+    labels_data[h0:, :h1, h2:] = 6
+    labels_data[h0:, h1:, :h2] = 7
+    labels_data[h0:, h1:, h2:] = 8
```

Comment:
Do you think we could we avoid this construction by using `generate_labeled_regions`?

https://github.com/nilearn/nilearn/blob/4f0ef229fbaa1ee32cd2eff352560381dc4ed8e2/nilearn/_utils/data_gen.py#L150-L164